### PR TITLE
schema: Add resources to system structure

### PR DESCRIPTION
**Pull request by jrha**
The resources structure is used by the broker to model children of a host
(or cluster) for example virtual machines on a hypervisor.

### CODE REVIEW

**Comment by jrha** [comment]
@urbonegi this is the schema change I was talking about at today's stand-up.

**Comment by jrha** [comment]
This is intimately related to the broker code that renders the plenaries:
https://github.com/quattor/aquilon/blob/b2e7a43d4dec16eb12ec21904166e12e8ed9ea38/lib/aquilon/worker/templates/resource.py#L157

**Comment by jrha** [comment]
Jenkins failing on `cannot locate template named 'quattor/aii/opennebula/schema'`

**Comment by jrha** [comment]
Naively it seems like the full list of keys under resources would be the following:
```
application
auto_start_list
filesystem
hostlink
intervention
reboot_iv
reboot_schedule
resourcegroup
service_address
share
system_list
virtual_machine
```

**Comment by urbonegi** [comment]
Looks good to me - the structure_resources is somethings that could be used in cluster archetypes for vm allocation.

**Comment by jrha** [comment]
Woo :green_heart:

File: quattor/types/aquilon.pan [hunk 1]
```diff
@@ -14,8 +14,24 @@ type structure_fsdata = {
...
 };
 
-type structure_resources = extensible {
-    "filesystem"    ? structure_fsdata[]
+type structure_virtual_machine = extensible {
+    "name" : string
+    "hardware" : structure_hardware
+};
+
+type structure_resources = {
```
Comment (stdweird) [inline quattor/types/aquilon.pan]: any chance for some annotation? in particaulr, waht is the `virtual_machine` used for?

Comment (jrha) [reply quattor/types/aquilon.pan]: If aquilon knows that a VM is on a particular hypervisor, that VM's hardware tree is copied into the hypervisor's profile under `/system/resources`

For example:
```
# ccm /system/resources/virtual_machine/1
+-/system/resources/virtual_machine/1
  +-hardware
    +-cards
      +-nic
        +-eth0
          $ boot : true
          $ hwaddr : 'aa:bb:cc:dd:ee:ff'
          $ media : 'Ethernet'
          $ name : 'Generic Network Interface'
          $ pxe : true
    +-cpu
      +-0
        $ arch : 'x86_64'
        $ cores : 1
        $ manufacturer : 'Intel'
        $ model : 'QEMU Virtual CPU version (cpu64-rhel6)'
        $ speed : 2000
      +-1
        $ arch : 'x86_64'
        $ cores : 1
        $ manufacturer : 'Intel'
        $ model : 'QEMU Virtual CPU version (cpu64-rhel6)'
        $ speed : 2000
    +-harddisks
      +-sda
        $ boot : true
        $ capacity : 10240
        $ interface : 'sata'
        $ model : 'Generic SATA disk'
        $ part_prefix : ''
    $ location : 'r89.harwell.europe'
    $ model : 'vm-openstack'
    $ nodename : 'vm-openstack-example-103'
    +-rack
      $ column : '420'
      $ name : 'r89rack420'
      $ room : 'example'
      $ row : '420'
    +-ram
      +-0
        $ size : 8192
    $ serialnumber : 'aaaabbbb-cccc-dddd-eeee-ffff00001111'
    +-sysloc
      $ building : 'r89'
      $ campus : 'harwell'
      $ city : 'harwell'
      $ continent : 'europe'
    $ type : 'singledisk'
  $ name : 'vm-openstack-example-103'
```

**Pull request merged**